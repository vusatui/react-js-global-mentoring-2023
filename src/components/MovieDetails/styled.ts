import styled from "styled-components";

export const StyledMovieDetails = styled.div`
  display: flex;
  flex-flow: row nowrap;
  align-items: flex-start;
  height: 486px;
`;

export const StyledMovieLeft = styled.div`
  width: 323px;
  min-width: 323px;
`;

export const StyledMovieImage = styled.img`
  width: 100%;
`;

export const StyledMovieRight = styled.div`
  display: flex;
  flex-flow: column;
  padding-left: 58px;
  width: 698px;
`;

export const StyledMovieRightTop = styled.div`
  padding-bottom: 30px;
`;

export const StyledMovieRightTopTop = styled.div`
  display: flex;
  flex-flow: row nowrap;
  margin-bottom: 9px;
  align-items: center;
`;

export const StyledMovieTitle = styled.div`
  margin-right: 25px;
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 300;
  font-size: 40px;
  line-height: 49px;
  letter-spacing: 1px;
  text-transform: uppercase;
  color: #FFFFFF;
`;

export const StyledMovieRating = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 60px;
  height: 60px;
  border: 1px solid #FFFFFF;
  border-radius: 50%;
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 300;
  font-size: 20px;
  line-height: 24px;
  text-align: center;
  text-transform: uppercase;
  color: #FFFFFF;
`;

export const StyledMovieGenres = styled.div`
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: #FFFFFF;
  mix-blend-mode: normal;
  opacity: 0.5;
`;

export const StyledMovieRightMiddle = styled.div`
  display: flex;
  flex-flow: row nowrap;
  margin-bottom: 30px;
`;


export const StyledMovieYear = styled.div`
  margin-right: 51px;
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 300;
  font-size: 24px;
  line-height: 29px;
  color: #F65261;
`;

export const StyledMovieDuration = styled.div`
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 300;
  font-size: 24px;
  line-height: 29px;
  color: #F65261;
`;

export const StyledMovieRightBottom = styled.div`
  height: 292px;
  overflow: auto;
`;

export const StyledMovieDescription = styled.div`
  font-family: 'Montserrat',serif;
  font-style: normal;
  font-weight: 300;
  font-size: 20px;
  line-height: 24px;
  color: #FFFFFF;
  mix-blend-mode: normal;
  opacity: 0.5;
`;

