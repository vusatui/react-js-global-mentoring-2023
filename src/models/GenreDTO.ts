export interface GenreDTO {
    name: string
}