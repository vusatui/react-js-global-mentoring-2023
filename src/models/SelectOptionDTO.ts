export interface SelectOptionDTO<T> {
    id: string
    label: string
    value: T
}