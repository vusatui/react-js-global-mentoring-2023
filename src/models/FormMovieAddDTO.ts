import {GenreDTO} from "./GenreDTO";

export interface FormMovieAddDTO {
    title: string
    movieUrl: string
    genre: GenreDTO[]
    overview: string
    runtime: string
    rating: string
    releaseDate: string
}