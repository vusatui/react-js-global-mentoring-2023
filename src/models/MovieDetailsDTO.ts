import {GenreDTO} from "./GenreDTO";

export interface MovieDetailsDTO {
    imageUrl: string
    title: string
    year: string
    genres: GenreDTO[]
    duration: string
    rating: number
    description: string
}