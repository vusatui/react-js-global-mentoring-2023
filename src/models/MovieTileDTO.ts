import {GenreDTO} from "./GenreDTO";

export interface MovieTileDTO {
    imageUrl: string
    title: string
    year: string
    genres: GenreDTO[]
}