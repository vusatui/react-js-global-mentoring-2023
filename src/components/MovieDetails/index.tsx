"use client";

import {useMemo} from "react";

import {MovieDetailsDTO} from "../../models/MovieDetailsDTO";

import {
    StyledMovieDescription,
    StyledMovieDetails, StyledMovieDuration, StyledMovieGenres,
    StyledMovieImage,
    StyledMovieLeft, StyledMovieRating,
    StyledMovieRight, StyledMovieRightBottom, StyledMovieRightMiddle,
    StyledMovieRightTop, StyledMovieRightTopTop,
    StyledMovieTitle, StyledMovieYear
} from "./styled";

const GENRE_SEPARATOR = ", ";

interface MovieDetailsProps {
    className?: string
    movieDetails: MovieDetailsDTO
}

const MovieDetails = ({
    className,
    movieDetails
}: MovieDetailsProps) => {
    const genres = useMemo(() => movieDetails.genres.map(genre => genre.name)
        .join(GENRE_SEPARATOR), [movieDetails]);

    return (
        <StyledMovieDetails className={className}>
            <StyledMovieLeft>
                <StyledMovieImage src={movieDetails.imageUrl} />
            </StyledMovieLeft>
            <StyledMovieRight>
                <StyledMovieRightTop>
                    <StyledMovieRightTopTop>
                        <StyledMovieTitle>
                            {movieDetails.title}
                        </StyledMovieTitle>
                        <StyledMovieRating>
                            {movieDetails.rating}
                        </StyledMovieRating>
                    </StyledMovieRightTopTop>
                    <StyledMovieGenres>
                        {genres}
                    </StyledMovieGenres>
                </StyledMovieRightTop>
                <StyledMovieRightMiddle>
                    <StyledMovieYear>
                        {movieDetails.year}
                    </StyledMovieYear>
                    <StyledMovieDuration>
                        {movieDetails.duration}
                    </StyledMovieDuration>
                </StyledMovieRightMiddle>
                <StyledMovieRightBottom>
                    <StyledMovieDescription>
                        {movieDetails.description}
                    </StyledMovieDescription>
                </StyledMovieRightBottom>
            </StyledMovieRight>
        </StyledMovieDetails>
    );
};

export default MovieDetails;