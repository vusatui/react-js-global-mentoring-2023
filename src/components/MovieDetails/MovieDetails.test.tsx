import {GenreDTO} from "../../models/GenreDTO";
import {MovieDetailsDTO} from "../../models/MovieDetailsDTO";
import MovieDetails from "./index";
import {render} from "@testing-library/react";

const genres: GenreDTO[] = [
    {name: "Drama"},
    {name: "Biography"},
    {name: "Music"},
];

const movieDetailsDTO: MovieDetailsDTO = {
    imageUrl: "https://upload.wikimedia.org/wikipedia/en/2/2e/Bohemian_Rhapsody_poster.png",
    title: "Bohemian Rhapsody",
    year: "2003",
    rating: 8.9,
    genres,
    duration: "2h 34min",
    description: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) " +
        "are two hit men who are out to retrieve a suitcase stolen from their employer, " +
        "mob boss Marsellus Wallace (Ving Rhames). " +
        "Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later " +
        "when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer " +
        "who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven " +
        "together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
};

const Component = () => (<MovieDetails movieDetails={movieDetailsDTO} />);

describe("MovieDetails", () => {

    it("Should renders correctly", () => {
        const { asFragment } = render(<Component />);
        expect(asFragment).toMatchSnapshot();
    });

});