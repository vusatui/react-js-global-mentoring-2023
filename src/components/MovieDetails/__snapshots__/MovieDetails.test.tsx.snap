// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Should renders correctly 1`] = `[Function]`;
